Jenkinsfile

```diff
@@ -127,310 +131,288 @@ pipeline {
-      }
-    }
-    stage('JDK Compatibility Tests') {
-      options { skipDefaultCheckout() }
-      when {
-        beforeAgent true
-        allOf {
-          expression { return env.ONLY_DOCS == "false" }
-          anyOf {
-            expression { return params.jdk_compatibility_ci }
-            expression { return env.GITHUB_COMMENT?.contains('jdk compatibility tests') }
-            expression { matchesPrLabel(label: 'ci:jdk-compatibility') }
-          }
-        }
-      }
-      matrix {
-        agent { label 'linux && immutable' }
-        axes {
-          axis {
-            // the list of support java versions can be found in the infra repo (ansible/roles/java/defaults/main.yml)
-            name 'JAVA_VERSION'
-            // 'openjdk18'  disabled for now see https://github.com/elastic/apm-agent-java/issues/2328
-            values 'openjdk17'
-
-          }
-        }
-        stages {
-          stage('Test') {
-            environment {
-              HOME = "${env.WORKSPACE}"
-              JAVA_HOME = "${env.HUDSON_HOME}/.java/${JAVA_VERSION}"
-              PATH = "${env.JAVA_HOME}/bin:${env.PATH}"
+          matrix {
+            agent { label 'linux && immutable' }
+            axes {
+              axis {
+                // the list of support java versions can be found in the infra repo (ansible/roles/java/defaults/main.yml)
+                name 'JAVA_VERSION'
+                // 'openjdk18'  disabled for now see https://github.com/elastic/apm-agent-java/issues/2328
+                values 'openjdk17'
+              }
             }
-            steps {
-              withGithubNotify(context: "Unit Tests ${JAVA_VERSION}", tab: 'tests') {
-                deleteDir()
-                unstashV2(name: 'build', bucket: "${JOB_GCS_BUCKET_STASH}", credentialsId: "${JOB_GCS_CREDENTIALS}") 
-                dir("${BASE_DIR}"){
-                  withOtelEnv() {
-                    sh(label: "./mvnw test for ${JAVA_VERSION}", script: './mvnw test')
+            stages {
+              stage('Test') {
+                steps {
+                  withGithubNotify(context: "Unit Tests ${JAVA_VERSION}", tab: 'tests') {
+                    deleteDir()
+                    unstashV2(name: 'build', bucket: "${JOB_GCS_BUCKET_STASH}", credentialsId: "${JOB_GCS_CREDENTIALS}") 
+                    dir("${BASE_DIR}"){
+                      withOtelEnv() {
+                        sh(label: "./mvnw test for ${JAVA_VERSION}", script: './mvnw test')
+                      }
+                    }
+                  }
+                }
+                post {
+                  always {
+                    reportTestResults()
                   }
                 }
-              }
-            }
-            post {
-              always {
-                reportTestResults()
               }
             }
           }
         }
       }
     }
-    stage('Stable') {
-      options { skipDefaultCheckout() }
+    stage('Releases'){
```

Comment:
Meta-stage for the release specific process, it runs on merge to master or tags. Each child-stage defines the when